oak_functions/examples/metrics/module/Cargo.toml

```diff
@@ -1,9 +1,12 @@
+cargo-features = ["per-package-target"]
+
 [package]
 name = "metrics"
 version = "0.1.0"
 authors = ["Conrad Grobler <grobler@google.com>"]
 edition = "2018"
 license = "Apache-2.0"
+default-target = "wasm32-unknown-unknown"
```

Comment:
This should probably be a forced-target?

Reply:
I would keep default-target. Or do do you see any reason to prohibit this to be overwritten by `cargo --target`?

Reply:
I think even in the cargo documentation this is the canonical example of when to use forced-target. A Wasm target is really meant to always be compiled to wasm. While other targets may be compiled e.g. for x86 or arm or whatever the host needs (hence it's expected they can be overriden).

Reply:
Sure, I would not worry too much about that, I don't think anyone is compiling natively for mac at the moment.

Reply:
So far I observed the following: 

If we force the `wasm32-unknown-unknown` target for e.g., key value lookup in its [`Cargo.toml`](https://github.com/project-oak/oak/blob/ccad8048c4cd6ebfc6bea04a6aa46dacc5abd16a/oak_functions/examples/key_value_lookup/module/Cargo.toml), for the corresponding (integration) [test](https://github.com/project-oak/oak/blob/ccad8048c4cd6ebfc6bea04a6aa46dacc5abd16a/oak_functions/examples/key_value_lookup/module/src/tests.rs) the call to `cargo test` will fail (as we try to compile the test with the `wasm` target).

I can think of two mitigations:
1. Move the test, e.g., as integration test to loader, so it gets a different `Cargo.toml`.
2. Don't force the target, but only set a `default-target` and with `cargo test` we can overwrite the target to the host target.

Can you see a different option?